### DISCUSSION TITLE
Problème a l'exécution de cypress:open

### DISCUSSION THREAD

**Discussion by Andrea-Graf** [web-classroom/web-classroom.github.io#44] [Q&A : Pratique / Labos]
Bonjour, j'ai un problème lors de l'utilisation de l'interface graphique de Cypress. Lorsque j'essaie de le lancer, un message d'erreur s'affiche, puis il ne se passe plus rien. Aucune fenêtre ne s'ouvre.

![image](https://github.com/web-classroom/web-classroom.github.io/assets/114007721/eb3d31a0-00d4-48b5-b3a4-9ab4ca180b43)

Le message d'erreur ne pose pas forcément problème, car lorsque je lance cypress:run, cela fonctionne, et le même message d'erreur s'affiche.

**Comment by hadrylouis** [comment]
Bonjour @Andrea-Graf ,

Pourrais-tu me montrer le contenu de ton fichier `package.json` qui se trouve à la racine du repo.

Merci

**Reply by Andrea-Graf** [reply]
Oui voici le contenue de mon fichier.
{
  "name": "01-html-css",
  "version": "1.0.0",
  "description": "Practice with HTML and CSS",
  "main": "index.js",
  "scripts": {
    "deploy": "./scripts/deploy.sh ./public",
    "dev": "live-server ./public --no-browser",
    "start": "serve ./public -l 8080",
    "cypress:open": "cypress open",
    "cypress:run": "cypress run",
    "test": "cypress run",
    "test-with-server": "start-server-and-test start http://localhost:8080"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tweb-classroom/1-html-css.git"
  },
  "author": "",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/tweb-classroom/1-html-css/issues"
  },
  "homepage": "https://github.com/tweb-classroom/1-html-css#readme",
  "devDependencies": {
    "cypress": "^10.8.0",
    "cypress-layout-inspector": "github:paulnta/cypress-layout-inspector#next",
    "live-server": "^1.2.2",
    "netlify-cli": "^11.8.3",
    "serve": "^14.0.1",
    "start-server-and-test": "^1.14.0",
    "wait-on": "^6.0.1"
  }
}

**Reply by hadrylouis** [reply]
Merci pour ces infos.

La commande `npm run cypress:open` va lancer Cypress avec son UI. J'observe que tu lances cette commande depuis WSL (ce qui est une bonne chose) mais le problème doit certainement venir de là. WSL ne semble pas arriver à ouvrir une fenêtre dans Windows depuis Linux.

Il m'est difficile de te donner une réponse qui corrige le problème. Il va falloir "debugger". Je pense que nous pourrons faire cela lors du prochain cours. Pourrais-tu déjà regarder la version de WSL que tu utilises (v1 ou v2) avec la commande `wsl --list --verbose`. Il faut s'assurer que tu aies bien WSL 2 qui supporte le lancement d'app Linux avec GUI (https://learn.microsoft.com/en-us/windows/wsl/tutorials/gui-apps)

Tu peux également vérifier que les dépendances Ubuntu/Debian pour Cypress sont bien installées dans ton WSL. Voir la réponse de @Nortalle https://github.com/orgs/web-classroom/discussions/42#discussioncomment-8618517

Si tu es bloqué dans ton travail, tu peux sans problème utiliser Cypress sans UI (l'output affiché est le même) via la commande `npm run cypress:run`
